extensions/common/vault/vault-gcp/src/test/java/org/eclipse/edc/vault/gcp/GcpSecretManagerVaultTest.java

```diff
@@ -0,0 +1,277 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.TestInstance.Lifecycle;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@TestInstance(Lifecycle.PER_CLASS)
+class GcpSecretManagerVaultTest {
+
+    class TestStatusCode implements StatusCode {
+        @Override
+        public Integer getTransportCode() {
+            return Integer.valueOf(0);
+        }
+
+        @Override
+        public Code getCode() {
+            return Code.OK;
+        }
+    }
+
+    private static final String TEST_REGION = "europe-west3";
+    private static final String TEST_PROJECT = "project";
+    private static final String VALID_KEY = "test";
+    private static final String INVALID_KEY = "test=";
+    private static final String SANITIZED_KEY = "test-_06924DEB";
+
+    private static final int MAX_KEY_LENGTH = 255;
+
+    private final Monitor monitor = mock(Monitor.class);
+    private final SecretManagerServiceClient secretClient = mock(SecretManagerServiceClient.class);
+    private final GcpSecretManagerVault vault = new GcpSecretManagerVault(monitor, TEST_PROJECT, TEST_REGION, secretClient);
+    private final Secret testSecret =
+                    Secret.newBuilder()
+                    .setReplication(
+                        Replication.newBuilder()
+                        .setUserManaged(Replication.UserManaged.newBuilder()
+                            .addReplicas(Replication.UserManaged.Replica.newBuilder()
+                                .setLocation(TEST_REGION)
+                                .build())
+                            .build())
+                        .build())
+                    .build();
+
+    private ArrayList<Character> validChars = new ArrayList<Character>();
+    private ArrayList<Character> invalidChars = new ArrayList<Character>();
+    private Random randGen = new Random();
+
+    private boolean isValidKey(String key) {
+        if (key.length() > MAX_KEY_LENGTH) {
+            return false;
+        }
+
+        for (int i = 0; i < key.length(); i++) {
+            var c = key.charAt(i);
+            if (!Character.isLetterOrDigit(c) && c != '_' && c != '-') {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private char getRandomValidChar() {
+        return validChars.get(randGen.nextInt(validChars.size()));
+    }
+
+    private char getRandomInvalidChar() {
+        return invalidChars.get(randGen.nextInt(invalidChars.size()));
+    }
+
+    private String getRandomKeyWithValidChars(int length) {
+        var sb = new StringBuilder();
+        for (int i = 0; i < length; i++) {
+            var c = getRandomValidChar();
+            sb.append(c);
+        }
+        return sb.toString();
+    }
+
+    private String getRandomKeyWithInvalidChars(int length, int errors) {
```

Comment:
as long as we know which are the invalid chars (e.g. `:`, `;` or so) why don't just create a string that contains one of these characters? this dynamic creation of a string with invalid characters seems unnecessarily complicated and prone to errors, better to keep tests as simple as possible.

Reply:
this is a bad practice, putting too much (untested) logic into a unit test brings into additional complexity and the possibility for false positives.